### PR TITLE
Ignore this PR

### PR DESCRIPTION
**Pull request by shangan23**
Added featured pages,

- PLP 
- PDP
- Cart

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { ProductDetailsComponent } from './components/shopping/products/product-details/product-details.component'
+import { ProductsComponent } from './components/shopping/products/products.component';
+import { ViewCartComponent } from './components/shopping/view-cart/view-cart.component';
+
+const routes: Routes = [
+  {
+    path: '',
+    component: ProductsComponent
```
Comment (ramsunvtech) [inline src/app/app-routing.module.ts]: this should have home page component not product

File: package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  "dependencies": {
+    "@angular/animations": "~10.1.5",
+    "@angular/common": "~10.1.5",
+    "@angular/compiler": "~10.1.5",
+    "@angular/core": "~10.1.5",
+    "@angular/forms": "~10.1.5",
+    "@angular/platform-browser": "~10.1.5",
+    "@angular/platform-browser-dynamic": "~10.1.5",
+    "@angular/router": "~10.1.5",
+    "bootstrap": "^4.5.3",
```
Comment (ramsunvtech) [inline package.json:22]: why cant u try material ?

File: src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    ShoppingComponent,
+    ProductsComponent,
+    ProductItemsComponent,
+    BreadcrumbComponent,
+    SpinnerComponent,
+    ProductDetailsComponent,
+    ViewCartComponent
+  ],
+  imports: [
+    BrowserModule,
```
Comment (ramsunvtech) [inline src/app/app.module.ts]: PLP, PDP, Checkout should be module similar to Browser Module

File: src/app/components/breadcrumb/breadcrumb.component.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<nav aria-label="breadcrumb">
+    <ol class="breadcrumb">
+        <li class="breadcrumb-item active" aria-current="page">
+                <a routerLink='/'>Home</a>
```
Comment (ramsunvtech) [inline src/app/components/breadcrumb/breadcrumb.component.html]: add `vscode` format config.

File: src/app/components/header/cart/cart-items/cart-items.component.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<button type="button" class="btn btn-primary btn-sm float-right text-center font-weight-light mx-auto" routerLink="/cart">
+    Cart ({{cartCount}}) <span *ngIf="cartTotal">- {{cartTotal | number | currency:'INR'}}</span>
```
Comment (ramsunvtech) [inline src/app/components/header/cart/cart-items/cart-items.component.html]: INR should come from config.

File: src/app/components/header/header.component.html [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<div class="container">
+    <div class="row sticky-top">
+        <div class="col-4">
+            <h2>Shopping Cart</h2>
+        </div>
+        <div class="col-6">
+            <!-- ul class="nav">
```
Comment (ramsunvtech) [inline src/app/components/header/header.component.html]: pls avoid pushing commented code

File: src/app/components/shopping/products/product-details/product-details.component.html [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<app-spinner *ngIf="loading"></app-spinner>
+<div class="row" *ngIf="!loading">
+    <div class="col">
+        <img class="img-thumbnail" style="width: auto; height: 450px;" [src]="product.image">
+    </div>
+    <div class="col">
+        <h3>{{product.title}}</h3>
+        <h6 class="display-6">{{product.price | currency:'INR'}}</h6>
+        <small class="text-muted">{{product.description}}</small>
+        <br/><br/><br/>
```
Comment (ramsunvtech) [inline src/app/components/shopping/products/product-details/product-details.component.html]: avoid multiple br

**Review by ramsunvtech — CHANGES_REQUESTED** [review]

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (shangan23) [reply src/app/app-routing.module.ts]: Added Home & NotFound component for serving home page and page not found.

File: package.json (hunk 2 above)
Comment (shangan23) [reply package.json:22]: Yes, we can. Maybe in the next version.

File: src/app/app.module.ts (hunk 3 above)
Comment (shangan23) [reply src/app/app.module.ts]: I have managed to keep PDP, PLP and Checkout under Shopping component but need to explore on creating a module with Angular.

File: src/app/components/breadcrumb/breadcrumb.component.html (hunk 4 above)
Comment (shangan23) [reply src/app/components/breadcrumb/breadcrumb.component.html]: Can you forward me to the link where I can find more information on this?

File: src/app/components/header/header.component.html (hunk 6 above)
Comment (shangan23) [reply src/app/components/header/header.component.html]: Removed comment.

File: src/app/components/shopping/products/product-details/product-details.component.html (hunk 7 above)
Comment (shangan23) [reply src/app/components/shopping/products/product-details/product-details.component.html]: removed and added a CSS class

File: src/app/components/header/cart/cart-items/cart-items.component.html (hunk 5 above)
Comment (shangan23) [reply src/app/components/header/cart/cart-items/cart-items.component.html]: done

File: src/app/app.module.ts (hunk 3 above)
Comment (shangan23) [reply src/app/app.module.ts]: Completed

File: src/app/components/breadcrumb/breadcrumb.component.html (hunk 4 above)
Comment (shangan23) [reply src/app/components/breadcrumb/breadcrumb.component.html]: Completed

File: src/app/app.module.ts [hunk 8]
```diff
@@ -1,44 +1,27 @@
...
-    NotFoundComponent
+    NotFoundComponent,
+    FooterComponent
   ],
   imports: [
-    BrowserModule,
-    AppRoutingModule,
-    HttpClientModule
+    ShopModule,
+    SharedModule,
```
Comment (ramsunvtech) [inline src/app/app.module.ts]: keep below modules.

```
GuestModule
Listingmodule
DetailModule
CartModule
```

Comment (shangan23) [reply src/app/app.module.ts]: Listingmodule - will it contain all lists like Products, Users, orders ... etc? and similarly DetailModule?

Why Shop module need to be split further?

Comment (shangan23) [reply src/app/app.module.ts]: Updated as per the comment

**Pull request closed without merging**